src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,45 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(actor);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add actor to DB: " + actor, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return actor;
     }
 
     @Override
     public Optional<Actor> get(Long id) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        Optional<Actor> actor;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            actor = Optional.ofNullable(session.get(Actor.class, id));
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't get actor from the DB with id: " + id, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return actor;
```

Comment:
You can use try-with-resources for get. You don't need to commit the transaction for reading operation.